src/Connection/Http1Connection.php

```diff
@@ -177,7 +177,7 @@ private function free(): Promise
 
     private function hasStreamFor(Request $request): bool
     {
-        $connectionUnlikelyToClose = $this->explicitTimeout && $this->getRemainingTime() > $this->timeoutGracePeriod;
+        $connectionUnlikelyToClose = $this->getRemainingTime() > $this->timeoutGracePeriod;
```

Comment:
`getRemainingTime()` will return `$this->timeoutGracePeriod`, which won't be greater than `$this->timeoutGracePeriod`, no?

Reply:
Oops, yes. What should the timeout be then if no explicit timeout is given?

Reply:
The hard connection close timeout will be `MAX_KEEP_ALIVE_TIMEOUT`. If the request is idempotent, the timeout is `MAX_KEEP_ALIVE_TIMEOUT` as well. If it is non-idempotent, it's `$this->timeoutGracePeriod`.

If there's an explicit timeout, it's `$remainingTime > $this->timeoutGracePeriod` for non-idempotent requests, idempotent requests don't need to check the remaining time.

Reply:
Yeah, ok, fixed. I should have been just comparing to 0 since I changed how `getRemainingTime()` worked.